<launch>

  <arg name="world_path" default="$(find uavros_gazebo)/worlds/lawn.world" />
  <arg name="vehicle" default="iris" />

  <include file="$(find uavros_gazebo)/launch/px4_gazebo_sitl.launch">
    <arg name="world" value="$(arg world_path)" />
    <arg name="vehicle" value="$(arg vehicle)" />   
  </include>

    <include file="$(find uavros_gazebo)/launch/mavros_px4.launch">
    </include>


    <node pkg="uavros_wrzf_sitl" type="uav_controller_node" name="uav_controller" output="screen">
        <param name="arrive_alt" value = "10.0"/>
        <param name="track_alt" value = "5.0"/>
        <param name="hover_yaw_rad" value = "0.75"/>
    	<param name="Kp" value = "1.0"/>
        <param name="Kd" value = "0.0"/>
        <param name="Ki" value = "0.0"/>
        <param name="vxy_max" value = "5.0"/>
        <param name="car_initposx" value = "3.0"/>
        <param name="car_initposy" value = "2.0"/>
    </node>

    <node pkg="uavros_wrzf_sitl" type="message_jc2fk.py" name="message_jc2fk" output="screen">
	<param name="txt_filepath" value = "$(find uavros_wrzf_sitl)/script/jc2fk.txt"/>
    </node>

  <arg name="visualization" default="false"/>
  <group if="$(arg visualization)">
      <node type="rviz" name="rviz" pkg="rviz" args="-d $(find uavros_uavugv_sitl)/config/rviz_single.rviz" />
  </group>

</launch>
